app/src/main/java/io/github/gmathi/novellibrary/fragment/WebPageDBFragment.kt

```diff
@@ -287,24 +296,27 @@ class WebPageDBFragment : BaseFragment() {
                     }
                 }
 
-                //Clean the document & Load the document on the webView
+                // Process the document and load it onto the webView
                 doc?.let { doc ->
                     val htmlHelper = HtmlHelper.getInstance(doc)
                     htmlHelper.removeJS(doc)
                     htmlHelper.additionalProcessing(doc)
                     htmlHelper.toggleTheme(dataCenter.isDarkTheme, doc)
 
                     if (dataCenter.enableClusterPages) {
-                        //Add the links-content to the doc
+                        // Get URL domain name of the chapter provider
+                        val baseUrlDomain = getUrlDomain(doc.location())
+
+                        // Add the content of the links to the doc
                         val hrefElements = doc.body().getElementsByTag("a")
                         hrefElements?.forEach {
                             if (it.hasAttr("href")) {
-
                                 val linkedUrl = it.attr("href")
                                 try {
-                                    val uri = Uri.parse(linkedUrl)
-                                    if (!HostNames.isItDoNotDownloadHost(uri.host)) {
-                                        val otherDoc = await { NovelApi().getDocumentWithUserAgent(it.attr("href")) }
+                                    // Check if URL is from chapter provider
+                                    val urlDomain = getUrlDomain(linkedUrl)
+                                    if (urlDomain == baseUrlDomain) {
+                                        val otherDoc = await { NovelApi().getDocumentWithUserAgent(linkedUrl) }
```

Comment:
Did you test this domain condition?

Reply:
Yeah, it only checks the top-level domain (ex. m.wuxiaworld.com or img.wuxiaworld.com and wuxiaworld.com are the same), so provided that the chapter provider serves their chapters on their domain (which would be weird if they didn't), it will work. I've tested it with a few and it works fine (better than before for me).